pkg/controllers/resources/storageclasses/host_syncer.go

```diff
@@ -51,6 +53,11 @@ func (s *hostStorageClassSyncer) Syncer() syncertypes.Sync[client.Object] {
 }
 
 func (s *hostStorageClassSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *synccontext.SyncToVirtualEvent[*storagev1.StorageClass]) (ctrl.Result, error) {
+	if !ctx.Config.Sync.FromHost.StorageClasses.Selector.Matches(event.Host) {
+		s.EventRecorder().Eventf(event.Host, "Warning", "SyncWarning", "did not sync storage class %q to virtual because it does not match the selector under 'sync.fromHost.storageClasses.selector'", event.Host.Name)
```

Comment:
We cannot use event.Host here as the event recorder only works for the virtual object